controllers/storageconsumer/storageconsumer_controller.go

```diff
@@ -174,6 +181,59 @@ func (r *StorageConsumerReconciler) reconcilePhases() (reconcile.Result, error)
 	return reconcile.Result{}, nil
 }
 
+func (r *StorageConsumerReconciler) createOnboardingSecret() error {
+	secret := &corev1.Secret{}
+	secret.Name = fmt.Sprintf("onboarding-token-%d", util.FnvHash(string(r.storageConsumer.UID)))
+	secret.Namespace = r.storageConsumer.Namespace
+	if err := r.get(secret); client.IgnoreNotFound(err) != nil {
+		return fmt.Errorf("failed to get secret %s: %v", secret.Name, err)
+	} else if secret.UID == "" {
+		if err := r.own(secret); err != nil {
+			return err
+		}
+		storageCluster, err := util.GetStorageClusterInNamespace(r.ctx, r.Client, r.namespace)
+		if err != nil {
+			return err
+		}
+		token, err := util.GenerateClientOnboardingToken(
+			r.TokenLifetimeInHours,
+			onboardingPrivateKeyFilePath,
+			storageCluster.UID,
+			r.storageConsumer.Name,
+		)
+		if err != nil {
+			return err
+		}
+		secret.Data = map[string][]byte{
+			"ticket": []byte(token),
+		}
+		util.AddAnnotation(secret, "token-expiry-utc", time.Now().Add(time.Duration(r.TokenLifetimeInHours)*time.Hour).UTC().String())
+		r.Log.Info("creating secret containing onboarding token for", "StorageConsumer", r.storageConsumer.Name, "Secret", secret.Name)
+		if err := r.Create(r.ctx, secret); err != nil {
+			return err
+		}
+	}
+	r.storageConsumer.Status.OnboardingTicketSecret = corev1.LocalObjectReference{
+		Name: secret.Name,
+	}
+	return nil
+}
+
+func (r *StorageConsumerReconciler) deleteOnboardingSecret() error {
+	secret := &corev1.Secret{}
+	secret.Name = fmt.Sprintf("onboarding-token-%d", util.FnvHash(string(r.storageConsumer.UID)))
+	secret.Namespace = r.storageConsumer.Namespace
+	if err := r.get(secret); client.IgnoreNotFound(err) != nil {
```

Comment:
There is not reason to get something just to delete it

Reply:
once enabled we always try to delete the secret and delete hits the api for which I saw a (similar) customer report stating their audit logs are filled w/ 404.

other way I thought of was to delete if not found in our status but that's reading our own status for taking a decision.

so, trying to get which hits cache and then hitting kube api if exists.

Reply:
as discussed, once the `.spec.enable` is set to true we need to delete secret which is a direct call to k8s api server throughout the storageconsumer (s) lifecycle which adds a log to api server w/ 404 errors as a side effect.

get is used to remove the side effect to the maximum possible, get hits cache first and slowly (let's say <5min) syncs the delete event as well, from then we'll not make any delete calls.

Reply:
deferred.